pkg/utils/sync_file_writer.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"io"
+	"os"
+	"sync"
+)
+
+var _ io.Writer = (*syncFileWriter)(nil)
+
+type syncFileWriter struct {
+	file *os.File
+	mux  sync.Mutex
+}
+
+// NewSyncFileWriter returns an io.Writer that is backed by an os.File
+// and that syncronizes the writes to the file.
+// This is used to serialize writes to os.Stdout for request logs.
```

Comment:
Explaining how it's used seems odd when exposing it in a public lib. :)

Perhaps just say that it is suitable for use with non-threadsafe writers, e.g. os.Stdout?

Reply:
Will do.